### ISSUE TITLE
CI: run go generate to ensure committed files are consistent with fs_vfsdata.go

### ISSUE THREAD

**Issue by myitcv** [gopherjs/gopherjs#684]
@shurcooL - unless I've totally misunderstood what's going on, at the moment it seems possible to commit a change to `compiler/natives/**/*.go` without re-running `go generate`, thereby leaving the committed `compiler/natives/fs_vfsdata.go` in a stale state, out of sync with its input files. 

Would it make sense to add a `go generate` step to the CI build and then verify that the committed `compiler/natives/fs_vfsdata.go` matches the CI-generated version (via something like `git status --porcelain`)?

**Comment by myitcv** [comment]
One slight issue is that the file modification times are also included in `compiler/natives/fs_vfsdata.go`; seems like that could be side-stepped by assuming the modification times of all virtual files was the start time of the VFS?

**Comment by shurcooL** [comment]
Yes, we should do something like this, but need to be mindful of modify times not matching.

We need to come up with a good way of making it happen.

**Comment by shurcooL** [comment]
One initial idea that comes to mind is that we don't necessarily need to ensure that the generated `fs_vfsdata.go` is zero diff vs what's committed. Instead, we can check that the **contents** of the committed [`natives.FS`](https://godoc.org/github.com/gopherjs/gopherjs/compiler/natives#FS) is what's supposed to be.

**Comment by shurcooL** [comment]
Also, I'll just point out, the value add of this is not very large. This kind of issue is quite easy to catch in code review. Even if missed, it'd likely cause a failure because the change has no effect. If not caught, it'd be caught by the next time natives are regenerated.

Of course, it'd still be nice to have CI catch it anyway.
